### DISCUSSION TITLE
add support for CycloneDX output of dependencies

### DISCUSSION THREAD

**Discussion by monperrus** [renovatebot/renovate#21911] [Suggest an Idea]
### Type of discussion.

I'm proposing an idea

### Tell us more.

Renovate is excellent at extracting all kind of dependencies, as can be seen in the "Dependency Dashboard' (for example https://github.com/INRIA/spoon/issues/4143)

Those dependencies form a complete software bill of material.

To a certain extent, Renovate is better at this than [Github's SBOM](https://github.blog/2023-03-28-introducing-self-service-sboms/) or [Google SBOM deps.dev](https://deps.dev/)

The only limitation is that there is no interoperable format.

It would be great to output the Renovate dependencies using the CycloneDX file format [cyclonedx.org/specification/overview](https://cyclonedx.org/specification/overview/)

**Comment by monperrus** [comment]
Per our conversation with @rarkins today, transitive dependencies are missing for that.

**Reply by monperrus** [reply]
And per the stateless nature of Renovate, the dependency dashboard is the source of truth, we may parse it.

**Comment by rarkins** [comment]
Renovate today tends to "not care about" transitive dependencies because it cares only about dependencies it can/should update. Possibly in the future this might need to change

**Comment by jamietanna** [comment]
FYI I've built[ a tool](https://dmd.tanna.dev/commands/renovate-to-sbom/) for this ([more info](https://www.jvt.me/posts/2023/11/03/renovate-to-sbom/)) that takes ie Renovate's debug logs and will then produce an SBOM - as above it'll only depend on the data that Renovate exposes, but can be a good start

**Reply by gberche-orange** [reply]
Thanks for sharing this work in 2024, this looks a very promising tool ! Is the tool still maintained ? 

Is there considerations to include some changes to renovate to make the SBOM exports easier and possibly more natively ?
